doc/decisions/0_drafts/keyset_types.md

```diff
@@ -0,0 +1,79 @@
+# Types of `KeySet`s
+
+## Problem
+
+A `KeySet` can be used in different ways.
+Among these are at least:
+
+- As originally intended, a set of configuration data
+- Metadata of a `Key`
+- General associative array data structure
+
+These types have slightly different requirements, and therefore may need to be treated differently.
+
+## Constraints
+
+- API duplication just to separate types should be avoided
```

Comment:
I agree about the goal but I don't think its a constraint.

```suggestion
- API must be minimal
```

Reply:
You could argue the API would still be minimal, since you cannot reduce it without loosing type safety or functionality.

Reply:
Maybe we make it an assumption:

"Introducing a type safe API with different C-types for each type of `KeySet`, is not worth the necessary API duplication"